### PR TITLE
AP_Notify: add ChibiOS ToneAlarm example

### PR DESCRIPTION
**Pull request by kd0aij**
enhancement part of https://github.com/ArduPilot/ardupilot/pull/8095 

This breaks PX4 builds, even though it's conditional on HAL_BOARD_CHIBIOS.

### CODE REVIEW

File: libraries/AP_Notify/examples/AP_ToneAlarm_Chibios_test/AP_ToneAlarm_ChibiOS_test.cpp [hunk 1]
```diff
@@ -0,0 +1,61 @@
+/*
+ *       Example of AC_Notify library .
+ *       DIYDrones.com
```
Comment (amilcarlucas) [inline libraries/AP_Notify/examples/AP_ToneAlarm_Chibios_test/AP_ToneAlarm_ChibiOS_test.cpp]: replace with ardupilot.org

File: libraries/AP_Notify/examples/AP_ToneAlarm_Chibios_test/AP_ToneAlarm_ChibiOS_test.cpp [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+// (state 3) -> (state 1)
+// (on every tick) if (complete) -> (state 0)
+
+void loop()
+{
+    static int tune = 0;
+    static uint32_t then = AP_HAL::millis();
+
+    hal.scheduler->delay(20);
+    uint32_t now = AP_HAL::millis();
```
Comment (amilcarlucas) [inline libraries/AP_Notify/examples/AP_ToneAlarm_Chibios_test/AP_ToneAlarm_ChibiOS_test.cpp]: const

**Comment by kd0aij** [comment]
@amilcarlucas Any ideas how to fix the CI failures?

**Comment by amilcarlucas** [comment]
No, I can not see why, maybe add some more commits to the PR to trigger a rebuild ?

**Comment by WickedShell** [comment]
This is failing CI, as it's attempting to build this on builds other then ChibiOS, but your checks are removing all the code from the non ChibiOS build. The easiest work around is to create a `#else` case for non ChibiOS builds that just does a panic immediately, or more effort (but correct) is to find a way to convince WAF to only build that example when working with a ChibiOS target.

**Comment by kd0aij** [comment]
@WickedShell Thanks, maybe it would be easier to make this a unit test, since that's it's actual purpose. One day I'll get around to learning how that's done in ArduPilot :)

**Comment by kd0aij** [comment]
@amilcarlucas There's a restart button on the "details" page for Travis. I think there's a similar mechanism for Semaphore also.

**Comment by WickedShell** [comment]
@kd0aij Yeah traivs just requires you to be signed into github, semaphore requires you to have authorized them with your github account (or signing in somehow) and then us adding you to the team for it. If you get an account on semaphore just ping me the user name and I will add you to the team so you can hit the button.

**Pull request closed without merging**